src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -248,8 +245,7 @@ public void Abort()
                         ResponseBody = _responseBody;
                         DuplexStream = new FrameDuplexStream(RequestBody, ResponseBody);
 
-                        _requestAbortCts = CancellationTokenSource.CreateLinkedTokenSource(_disconnectCts.Token);
-                        RequestAborted = _requestAbortCts.Token;
+                        RequestAborted = _disconnectOrAbortedCts.Token;
```

Comment:
I think this is incorrect. RequestAborted needs to be unique per request, not per connection. We don't want registrations from one request to fire for the next request.